packages/toolkit/utils/src/cli/patch.ts

```diff
@@ -0,0 +1,28 @@
+/* eslint-disable eslint-comments/disable-enable-pair */
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+import path from 'path';
+import { pathToFileURL } from 'url';
+
+const GLOBAL_PATCHED_SYMBOL = Symbol('GLOBAL_PATCHED_SYMBOL');
+
+/** fix issue about dart2js: https://github.com/dart-lang/sdk/issues/27979 */
+export function patchGlobalLocation() {
+  const href = `${pathToFileURL(process.cwd()).href}${path.sep}`;
+  // @ts-expect-error
+  href[GLOBAL_PATCHED_SYMBOL] = true;
+  // @ts-expect-error
+  global.location ||= Object.freeze({ [GLOBAL_PATCHED_SYMBOL]: true });
+  global.location.href ||= href;
+}
+
+export function unpatchGlobalLocation() {
+  // @ts-expect-error
```

Comment:
Why so many ts-expect-error?

Reply:
Because Typescript prevents me from modifying the global object.

Reply:
why not using `globalThis`

Reply:
I think they are equal.
![image](https://github.com/web-infra-dev/modern.js/assets/18379948/9c5a42aa-4a4f-4edb-87b2-dbbc689afa5b)